### DISCUSSION TITLE
Redirect_uri_mismatch using Oauth router

### DISCUSSION THREAD

**Discussion by aidaph** [fastapi-users/fastapi-users#1164] [Q&A]
Hello, 

I'm trying to authenticate my API users using the github oauthenticator (https_oauth github client). I want to get the access token when callback. Likewise, I followed the whole example of the docs for the oauth applications, changing the google oauth by github. Apart from that and to not let the api choose, the redirect url is added as parameter of the get_oauth_router call. In github, the oauth dev app is configured with the same url as the defined in this call (http://example.com/callback). Finally, instead of using the `Oauth2PasswordBearer`  Fastapi specification, I've changed it to `OAuth2AuthorizationCodeBearer`  as the code below to get the access token without an username and password. Is it possible? When I `authorize` my app against github using the corresponding buttom, the github callback result in the error `redirect_uri_mismatch`. 

main-app.py
```
REDIRECT_URL = "http://example.com/callback"
app.include_router(
    fastapi_users.get_oauth_router(
        github_oauth_client, 
        auth_backend, 
        SECRET, ## PASSWORD TO CHANGE
        associate_by_email=True,
        redirect_url=f"{REDIRECT_URL}"),
    prefix="",
    tags=["auth"],
)
```

user.py
```
bearer_transport = BearerTransport(tokenUrl="/auth/jwt/login")
bearer_transport.scheme = OAuth2AuthorizationCodeBearer(authorizationUrl="https://github.com/login/oauth/authorize", 
                                                        tokenUrl="https://github.com/login/oauth/access_token",
                                                        scopes={"profile": "profile", "openid": "openid"})
```

Debugging the code, the authorization url is the following. As I can see, although the redirect uri is urlencoded is the same configured in the github oauth developer console.
`https://github.com/login/oauth/authorize?response_type=code&client_id=XXXXXXXXXX&redirect_uri=http%3A%2F%example.com%2Fcallback&state=XXXXXXXXXXXEchWouQdPH_B_oXXXXXXX&scope=user+user%3Aemail`

What am I missing in the code? 

Thanks in advance!

**Comment by frankie567** [comment]
Hi @aidaph 👋 

From what I understand, you're trying to do an OAuth2 authentication *outside* the context of FastAPI, by using the Swagger `Authorize` button with the `OAuth2AuthorizationCodeBearer`. It could work, but it's a completely different way of thinking from FastAPI Users.

If you want to make OAuth2 with FastAPI Users (with complete user registration on your DB), you shouldn't modify the scheme. Then, you should manually call the `/authorize` API and then process the `/callback`: https://fastapi-users.github.io/fastapi-users/10.4/usage/routes/#oauth-router
